src/modules/esl-utils/misc/test/array.test.ts

```diff
@@ -45,3 +45,40 @@ describe('misc/array helper tests', () => {
     expect(range(9, (x) => x / 8)).toEqual([...Array(9).keys()].map((x) => x / 8))
   });
 });
+
+test('intersection', () => {
```

Comment:
Please also check edge cases:
Currently method signature accepts 0 or 1 argument, find the way to disallow such cases or make intersect return an empty array or the first argument.
The same applicable to other methods